### ISSUE TITLE
ESP32 Compatibility

### ISSUE THREAD

**Issue by westonkno3it** [foxis/ESPReflow#5]
When building for ESP32 I'm getting errors regarding SPIFFS:
src\Config.cpp: In member function 'bool Config::load_json(const String&, size_t, Config::THandlerFunction_parse)':
src\Config.cpp:132:20: error: 'SPIFFS' was not declared in this scope
File configFile = SPIFFS.open(name, "r");
^

src\main.cpp:171:2: error: 'SPIFFS' was not declared in this scope
SPIFFS.begin();
^

src\main.cpp: In lambda function:
src\main.cpp:184:61: error: 'SPIFFS' is not captured
AsyncWebServerResponse *response = request->beginResponse(SPIFFS, "/profiles.json");
^

src\main.cpp: In lambda function:
src\main.cpp:192:61: error: 'SPIFFS' is not captured
AsyncWebServerResponse *response = request->beginResponse(SPIFFS, "/config.json");
^

Also tone() isn't written for ESP32 AFAIK:
src\ControllerBase.cpp:72:25: error: 'tone' was not declared in this scope
tone(BUZZER_A, 440, 100);
^

No problem getting this to build with ESP8266 set under platformIO. Just had to get the older version of ArduinoJSON. Single header downloaded from here: https://arduinojson.org/v5/doc/installation/

**Comment by foxis** [comment]
I never tried to build it for ESP32, let alone run it. I have some vague thoughts on upgrading the board to ESP32 in the future, but that may happen after I put it out on Tindie and see what people actually need :)

As for the ArduinoJSON library, I'll port the code to newer library as soon as possible.

**Comment by westonkno3it** [comment]
I managed to get it to build with some issues afterwards. 

Here's an incomplete list of things I've found out:
SPIFFS.h needs to be included in Config.cpp & main.cpp

the WiFi.setHostname() method in EasyOTA.cpp doesn't accept raw strings. I just created a character buffer, copied the data, and passed that to the function. Probably better ways to do it but it did work

In the switch statement right below the setHostname() section i had to prepend all the enum values with their type e.g. STATE::EOS_SETUP_WIFI.

WiFi.getNetworkInfo() in EasyOTA.cpp doesnt take the bool hidden_scan on the end. I commented out that line and just used the same function omitting that parameter.

Also changed uin8 bestBSSID[6] to uint8_t

The biggest problem I ran into was it would crash almost immediately after booting. I traced where the issue was occurring and it was trying to get a tcp connection but was not connected to wifi. I added 
config.OTA->loop();
to the begging of setup() to force it to connect before calling server.begin. Also i had changed the variables in wificonfig.h to the correct values but it didn't seem to help. Adding an entry in config.json for networks did but I didn't see instructions for that anywhere.

Once it compiled i was able to connect to the web-sever was able to start a profile. However, it would immediately timeout from watchdog timer. I've checked the code and don't fully understand why that is occurring. As a temporary work around I have disabled the watchdog timer and just keep a closer eye on it.

Not sure if this info helps. Just thought I would share what I've figured out. Would love to see this on Tindie. I'm currently trying to figure out how to calibrate the device to my Toaster oven but I think I'll be able to manage. Thank you for releasing this! It's a great project!
